src/meta/client/MetaClient.cpp

```diff
@@ -286,6 +286,31 @@ MetaClient::reverse(const PartsAlloc& parts) {
     return hosts;
 }
 
+template<typename Request,
+         typename RemoteFunc,
+         typename RespGenerator,
+         typename RpcResponse,
+         typename Response>
+void MetaClient::retry(Request req,
+                       RemoteFunc remoteFunc,
+                       RespGenerator respGen,
+                       folly::Promise<StatusOr<Response>> pro,
+                       bool toLeader,
+                       int32_t retry,
+                       int32_t retryLimit) {
+    folly::EventBase retryEvb;
+    auto observer = folly::AsyncTimeout::make(retryEvb,
+            [req = std::move(req), remoteFunc = std::move(remoteFunc),
+             respGen = std::move(respGen), pro = std::move(pro), toLeader, retry,
+             retryLimit, this] () mutable noexcept {
+        getResponse(std::move(req), std::move(remoteFunc), std::move(respGen),
+                    std::move(pro), toLeader, retry + 1, retryLimit);
+    });
+    observer->scheduleTimeout(
+            std::chrono::milliseconds(FLAGS_meta_client_retry_interval_secs * 1000));
+    retryEvb.loop();
```

Comment:
Is it still blocking the thread?

Reply:
Seems so.

Reply:
Yeah, it would block the thread, but it is a stack allocated evb, it won't block our io thread pool